src/EditorFeatures/TestUtilities2/Intellisense/ModernCompletionTestState.vb

```diff
@@ -183,21 +183,30 @@ Namespace Microsoft.CodeAnalysis.Editor.UnitTests.IntelliSense
             Task.WaitAny(task1, task2)
         End Sub
 
-        Public Overrides Async Function AssertCompletionSession(Optional projectionsView As ITextView = Nothing) As Task
+        Public Overrides Function AssertCompletionSession(Optional projectionsView As ITextView = Nothing) As Task
             Dim view = If(projectionsView, TextView)
 
-            Await WaitForAsynchronousOperationsAsync()
             Dim session = GetExportedValue(Of IAsyncCompletionBroker)().GetSession(view)
             Assert.True(session IsNot Nothing, "AssertCompletionSession")
+            Return Task.FromResult(0)
         End Function
 
+        Public Overrides Sub AssertCompletionItemsDoNotContainAny(displayText As String())
+            Dim items = GetCompletionItems()
+            Assert.False(displayText.Any(Function(v) items.Any(Function(i) i.DisplayText = v)))
+        End Sub
+
+        Public Overrides Sub AssertCompletionItemsContainAll(displayText As String())
+            Dim items = GetCompletionItems()
+            Assert.True(displayText.All(Function(v) items.Any(Function(i) i.DisplayText = v)))
+        End Sub
+
         Public Overrides Async Function AssertCompletionSessionAfterTypingHash() As Task
             ' starting with the modern completion implementation, # is treated as an IntelliSense trigger
             Await AssertCompletionSession()
         End Function
 
         Public Overrides Sub AssertItemsInOrder(expectedOrder As String())
-            AssertNoAsynchronousOperationsRunning()
```

Comment:
❔ Why remove this? Applies to all such cases.

Reply:
AssertNoAsynchronousOperationsRunning() was usually called after WaitForAsynchronousOperationsAsync(). The Wait waits for processed not used in the modern completion. The Assert check that those processes are not currently running. Both seem to be useless in the modern completion.

---
In reply to: [314786131](https://github.com/dotnet/roslyn/pull/38021#discussion_r314786131) [](ancestors = 314786131)